justfile

```
@@ -162,22 +162,23 @@ default_db_url := 'postgresql://' + db_user + ':' + db_password + '@' + db_host
 database_url := env_var_or_default('DATABASE_URL', default_db_url)
 default_migration_params := ''
 v2_migration_dir := source_directory() / 'v2_migrations'
+v2_compatible_migrations := source_directory() / 'v2_compatible_migrations'
 v1_migration_dir := source_directory() / 'migrations'
 resultant_dir := source_directory() / 'final-migrations'
 
-# Copy v1 and v2 migrations to a single directory
+# Copy migrations in {{dir_1}} and {{dir_2}} to a single directory {{resultant_dir}} after prefixing the subdirectories of {{dir_2}} with {{prefix}}
 [private]
-copy_migrations:
+prefix_and_copy_migrations dir_1 dir_2 prefix resultant_dir:
     #! /usr/bin/env bash
     mkdir -p {{resultant_dir}}
-    cp -r {{v1_migration_dir}}/* {{resultant_dir}}/
+    cp -r {{dir_1}}/* {{resultant_dir}}/ > /dev/null 2>&1
 
-    # Prefix v2 migrations with 9
+    # Prefix v2 migrations with {{prefix}}
     sh -c '
-    for dir in "{{v2_migration_dir}}"/*; do
+    for dir in "{{dir_2}}"/*; do
         if [ -d "$dir" ]; then
             base_name=$(basename "$dir")
-            new_name="9$base_name" 
+            new_name="{{prefix}}$base_name" 
             cp -r "$dir" "{{resultant_dir}}/$new_name"
```

Nit: Using `{}` around shell variables.

```suggestion
        if [ -d "${dir}" ]; then
            base_name=$(basename "${dir}")
            new_name="{{prefix}}${base_name}" 
            cp -r "${dir}" "{{resultant_dir}}/${new_name}"
```